scripts/core/editor3/components/links/LinkInput.tsx

```diff
@@ -80,13 +82,19 @@ export class LinkInputComponent extends React.Component<any, any> {
      */
     onSubmit(linkType) {
         let link;
-        const {suggestingMode} = this.props;
+        const {suggestingMode, localDomains} = this.props;
         const _createLinkSuggestion = this.props.createLinkSuggestion;
         const _applyLink = this.props.applyLink;
         const _changeLinkSuggestion = this.props.changeLinkSuggestion;
 
         if (linkType === linkTypes.href) {
-            link = {href: this.state.url};
+            const url = this.state.url;
+            const isLocalDomain = (localDomains || []).some((item) => url.includes(item.domain));
+
+            link = {href: url};
+            if (!isLocalDomain && localDomains != null) {
+                link.target = '_blank';
```

Comment:
I think it'd be better to use a custom attribute like `data-sdexternal` instead of a `target`. Target is a display preference and we shouldn't couple it with data.